### PR TITLE
Identify additional path method args and include in resource action para...

### PR DESCRIPTION
**Pull request by mattmccabe**
WIP fixing Issue #78
@bajtos let me know if I am heading in the right direction. I tried to somewhat pattern after how 'isUser' is  added to the method in services.js and then keyed off of it in the ejs template.

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by bajtos** [comment]
@slnode ok to test

**Comment by bajtos** [comment]
Thank you for the pull request. The implementation looks good in general, let me know when you consider it done.

Please rewrite the commit message to follow the [50/72 rule](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html). If you are not familiar with git history manipulation, you need `git rebase -i master` and `git push -f`.

**Comment by mattmccabe** [comment]
Thanks for the rewrite/rebase instructions, I was not previously familiar with the interactive rebase. I have pushed the new commit message.

From the contribution guide, I think I have everything covered to consider this done. Let me know if there is anything I missed.

File: lib/services.js [hunk 1]
```diff
@@ -62,6 +62,21 @@ function describeModels(app) {
       var ctor = method.restClass.ctor;
       if (!ctor || method.sharedMethod.isStatic) return;
       method.accepts = ctor.accepts.concat(method.accepts);
+
+      if(!method.accepts) return;
+      //Any extra http action arguments in the path need to be added to the
+      //angular resource actions as params
```
Comment (bajtos) [inline lib/services.js]: Coding style:

 - please add an empty line after the return statement to visually delimit different functional blocks of code
 - add a space after trailing slash - `// Any`, `// angular`

**Comment by bajtos** [comment]
> Fixes #78, add params to resource action

Use imperative form and don't include bug number in the commit summary: "Add params to resource actions".

Also add unit-tests to verify that the implementation works as intended and prevent regressions in the future. See my comment in #78:

> Don't forget to add a unit-test verifying your implementation. We already have some tests for hasAndBelongsToMany relation, just add a new test to check the method link.

> https://github.com/strongloop/loopback-sdk-angular/blob/fa7668b8e11e04fad8d239558846804e5d6315f8/test.e2e/spec/services.spec.js#L580-L701

**Comment by mattmccabe** [comment]
I saw the examples for the hasAndBelongsToMany, and added my own. I followed the setup instructions in the [DEVELOPING.md](https://github.com/strongloop/loopback-sdk-angular/blob/master/DEVELOPING.md) but when I run my test for linking and unlinking I get a 404 error. I am using the same Product and Category models as the other hasAndBelongsToMany tests so I didn't think there would be an additional setup needed in the test-server. Any quick pointers on the puzzle--piece I may be missing?

**Comment by bajtos** [comment]
Can you please push the test code that is returning 404, so that I can run and debug it on my machine?

**Comment by mattmccabe** [comment]
Pushed the test code, and the style guide corrections. Still getting used to this environment so it could be something quite pedestrian that I am overlooking.

**Comment by bajtos** [comment]
I have fixed the failing tests myself, made few coding-style related improvements and landed the patch via 71b00dd. Thank you for the contribution!

The tests were failing because you forgot to return the first promise created in the test. By the time your test got to call link/unlink, another test suite was already running and a different set of models was configured on the server.

File: test.e2e/spec/services.spec.js [hunk 2]
```diff
@@ -680,6 +680,34 @@ define(['angular', 'given', 'util'], function(angular, given, util) {
...
+            expect(list.length).to.eql(0);
+          });
+        }, function(fault) {
+          console.log('unlinking failed', fault);
+        });
+      });
+
+      it('links a model', function() {
+        var cat = Category.create({ name: 'linked-cat' });
+        cat.$promise.then(function() {
```
Comment (bajtos) [inline test.e2e/spec/services.spec.js]:
```js
return cat.$promise.then //...
```

File: test.e2e/spec/services.spec.js [hunk 3]
```diff
@@ -680,6 +680,34 @@ define(['angular', 'given', 'util'], function(angular, given, util) {
           });
       });
 
+      it('unlinks a model', function() {
+        var unlnk = Product.categories.unlink(
+          { id: testData.product.id, fk: testData.category.id});
+        unlnk.$promise.then(function(data) {
```
Comment (bajtos) [inline test.e2e/spec/services.spec.js]:
```js
return unlnk.$promise.then //...
```

**Comment by mattmccabe** [comment]
Glad I was able to help. Thanks for all of the pointers and assistance.

**Pull request closed without merging**